evidence_test.go

```diff
@@ -167,3 +168,50 @@ func TestEvidence_Verify_no_message(t *testing.T) {
 	err := evidence.Verify(pk)
 	assert.EqualError(t, err, "no Sign1 message found")
 }
+
+func TestEvidence_p2_sign_and_verify_key_mismatch(t *testing.T) {
```

Comment:
Is it important for this test that the claims-set is p2?  If not, just drop the p2 from the test.  Otherwise there should be a similar test for p1 too.

Reply:
Removed p2 as it is not really checking anything specific to p2 and does not add any value to duplicate the test for p1